spikeextractors/multirecordingchannelextractor.py

```diff
@@ -35,21 +35,21 @@ def __init__(self, recordings, groups=None):
         # set group information for channels if available
         if groups is not None:
             if len(groups) == len(recordings):
+                group_values = []
                 for i, group in enumerate(groups):
                     recording = recordings[i]
                     channel_ids = recording.get_channel_ids()
-                    recording.set_channel_groups(groups=np.repeat(group, len(channel_ids)), channel_ids=channel_ids)
+                    group_values += [group] * len(channel_ids)
+                self.set_channel_groups(groups=group_values)
```

Comment:
This sets groups and locations for the multirecording object directly, without affecting the recordings in the recording list

Reply:
Yep better